loony.nim

```diff
@@ -208,6 +208,7 @@ proc advHead(queue: LoonyQueue; curr, h, t: var TagPtr): AdvHead =
 
 proc pushImpl[T](queue: LoonyQueue[T], el: T,
                     forcedCoherance: static bool = false) =
+  assert not queue.isNil(), "The queue has not been initialised"
```

Comment:
We need to read the reference anyway, so we're just paying the cost of a conditional.  I think we should make this check even in `release` or `danger` builds.

Reply:
Fair; I wasn't sure what the cost would be. I guess it's also significantly less than all the atomic writes lol